### PR TITLE
Minor tweak to contrib docs

### PR DESCRIPTION
**Pull request by cijothomas**

### CODE REVIEW

File: CONTRIBUTING.md
Comment (reyang) [inline CONTRIBUTING.md:170]:
```suggestion
release and the maintainers decided to defer the PR to the next release train.
```

File: CONTRIBUTING.md
Comment (reyang) [inline CONTRIBUTING.md:169]:
```suggestion
PRs may not be merged immediately if the repo is in the process of a
```

**Review by reyang — APPROVED** [review]

**Comment by cijothomas** [comment]
Will wait to get all @open-telemetry/dotnet-maintainers to approve/comment before merging.

**Review by mic-max — APPROVED** [review]

**Review by utpilla — APPROVED** [review]

**Review by alanwest — APPROVED** [review]

**Pull request merged**
